[python3] Python3 needs Pyhon3 to build Python3

Till four days ago, it was possible to install Python3 in vcpkg without having Python3 system-wide installed. Since 4 days ago (and specifically, this commit and line: https://github.com/microsoft/vcpkg/commit/c6928dfb9eb7f333f988d2ce70d1e771b7606849#diff-f682e6e46ef0d0dae2ceba81d191975da58a84cbdab4811a0e84683442265461R245 ) that is no longer the case.

We use vcpkg to build a linux-generic binary with a very old libc so as many people as possible can run it on their system. Basically, it based on `quay.io/pypa/manylinux2014_x86_64`. Sadly, having a system-wide Python3 on that is very challenging, one that vcpkg was resolving for us. But .. no longer :(

Would be really appreciated if Python3 can be build again without needing Python3 to start with! Tnx :)

Package: python3:x64-linux -> 3.11.4#1

**Host Environment**

- Host: x64-linux
- Compiler: GNU 10.2.1
-    vcpkg-tool version: 2023-08-09-9990a4c9026811a312cb2af78bf77f3d9d288416
    vcpkg-scripts version: 0fa8459 2023-08-11 (34 hours ago)

**To Reproduce**

`vcpkg install python3`
**Failure logs**

```
-- Downloading https://github.com/python/cpython/archive/v3.11.4.tar.gz -> python-cpython-v3.11.4.tar.gz...
-- Extracting source /vcpkg/downloads/python-cpython-v3.11.4.tar.gz
-- Applying patch 0001-only-build-required-projects.patch
-- Applying patch 0003-use-vcpkg-zlib.patch
-- Applying patch 0004-devendor-external-dependencies.patch
-- Applying patch 0005-dont-copy-vcruntime.patch
-- Applying patch 0008-python.pc.patch
-- Applying patch 0010-dont-skip-rpath.patch
-- Applying patch 0012-force-disable-curses.patch
-- Applying patch 0014-fix-get-python-inc-output.patch
-- Applying patch 0015-dont-use-WINDOWS-def.patch
-- Applying patch 0002-static-library.patch
-- Applying patch 0011-gcc-ldflags-fix.patch
-- Using source at /vcpkg/buildtrees/python3/src/v3.11.4-101b2af427.clean
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:163 (message):
  Could not find python3.  Please install it via your package manager:

      sudo apt-get install python3
Call Stack (most recent call first):
  ports/python3/portfile.cmake:254 (vcpkg_find_acquire_program)
  scripts/ports.cmake:147 (include)
```

--- a/ports/python3/portfile.cmake
+++ b/ports/python3/portfile.cmake
@@ -250,9 +250,6 @@ else()
     if(VCPKG_CROSSCOMPILING)
         set(_python_for_build "${CURRENT_HOST_INSTALLED_DIR}/tools/python3/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
         list(APPEND OPTIONS "--with-build-python=${_python_for_build}")
-    else()
-        vcpkg_find_acquire_program(PYTHON3)
-        list(APPEND OPTIONS "ac_cv_prog_PYTHON_FOR_REGEN=${PYTHON3}")
     endif()
 
     vcpkg_configure_make(

--- a/ports/python3/vcpkg.json
+++ b/ports/python3/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "python3",
   "version": "3.11.4",
-  "port-version": 1,
+  "port-version": 2,
   "description": "The Python programming language",
   "homepage": "https://github.com/python/cpython",
   "license": "Python-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6602,7 +6602,7 @@
     },
     "python3": {
       "baseline": "3.11.4",
-      "port-version": 1
+      "port-version": 2
     },
     "qca": {
       "baseline": "2.3.5",

--- a/versions/p-/python3.json
+++ b/versions/p-/python3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "4f49a2d2249cb9f81e3a5c83329f383de36faf16",
+      "version": "3.11.4",
+      "port-version": 2
+    },
+    {
       "git-tree": "531061b4485adebb80364473d227b0edf4ee8ed9",
       "version": "3.11.4",
       "port-version": 1

